.github/workflows/integration_test_allure.md

```diff
@@ -0,0 +1,64 @@
+# Allure Report Integration
+
+[Allure Report](https://allurereport.org/) for [integration_test_run.yaml](https://github.com/canonical/operator-workflows?tab=readme-ov-file#integration-test-workflow-canonicaloperator-workflowsgithubworkflowsintegration_testyamlmain)
+
+In order to integrate Allure with your repository, perform the following actions:
+
+## 1. Adding allure-pytest and pytest collection plugin
+
+Please add the following into the `requirements.txt` that is called by the integration test -
+
+```
+allure-pytest>=2.8.18
+```
+
+Add the following line under the dependencies (`deps`) in the integration section inside `tox.ini` -
+
+```
+git+https://github.com/canonical/operator-workflows@main\#subdirectory=python/pytest_plugins/allure_pytest_collection_report
+```
+
+## 2. Calling the allure-workflow
+
+Add the following lines at the end of the workflow that runs the integrations tests by calling the reuable workflow [integration-test.yaml](https://github.com/canonical/operator-workflows/blob/main/.github/workflows/integration_test.yaml):
+
+```
+  allure-report:
+    if: always() && !cancelled()
+    needs:
+      - [list of jobs with tests you would like to visualize]
+    uses: canonical/operator-workflows/.github/workflows/allure_report.yaml@main
+```
+
+Here's an [example for the above](https://github.com/canonical/github-runner-operator/pull/412).
+
+**NOTE:** If the workflow is being called inside a matrix with the same test modules run with different parameters, the allure report will only display the results of the last combination.
+
+## 3. Changing branch permissions
+
+**NOTE:** For this, you would require admin access to the repository.
+
+- Go to the repository's **Settings > Branches** and next to Branch protection rules, select **Add rule**
+- Enter the branch name **gh-pages** and select **Allow force pushes** and click **Save changes**
+
+## 4. Github pages branch
+
+- Create `gh-pages` branch:
+
+```
```

Comment:
Could this segment be automated as well so the repos can start using this right away?

Reply:
I believe the separate protection rule is no longer required? Hence we could check if the branch is present, continue if it is, create one if it isn't :)

Reply:
by default for all the repos we have a global protection rule for all branches requiring signed commits. We require a seperate protection rule for gh-pages branches with the signed commits unchecked otherwise the last step in the allure_report workflow fails to push the changes into that branch.

Reply:
<img width="792" alt="image" src="https://github.com/user-attachments/assets/c943da86-08f8-4d5b-a607-5d5f49477211">
this is the global rule i was referring to